app/dao/fact_notification_status_dao.py

```diff
@@ -242,6 +243,105 @@ def get_last_send_for_api_key(api_key_id):
     ).all()
 
 
+def get_api_key_ranked_by_notifications_created(n_days_back):
```

Comment:
This is a monster query with case statements. Basically I'm getting the total email and sms notifications created for each api key. I could have moved more of this logic into python, but I figured SQL would be faster and there are other ugly queries in this file with `CASE` statements, like `fetch_monthly_notification_statuses_per_service` on line 595.